homeassistant/components/zha/core/store.py

```diff
@@ -0,0 +1,146 @@
+"""Data storage helper for ZHA."""
+import logging
+from collections import OrderedDict
+# pylint: disable=W0611
+from typing import MutableMapping  # noqa: F401
+from typing import cast
+
+import attr
+
+from homeassistant.core import callback
+from homeassistant.loader import bind_hass
+from homeassistant.helpers.typing import HomeAssistantType
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_REGISTRY = 'zha_storage'
+
+STORAGE_KEY = 'zha.storage'
+STORAGE_VERSION = 1
+SAVE_DELAY = 10
+
+
+@attr.s(slots=True, frozen=True)
+class ZhaDeviceEntry:
+    """Zha Device storage Entry."""
+
+    name = attr.ib(type=str, default=None)
+    ieee = attr.ib(type=str, default=None)
+    power_source = attr.ib(type=int, default=None)
+    manufacturer_code = attr.ib(type=int, default=None)
+
+
+class ZhaDeviceStorage:
+    """Class to hold a registry of zha devices."""
+
+    def __init__(self, hass: HomeAssistantType) -> None:
+        """Initialize the zha device storage."""
+        self.hass = hass
+        self.devices = {}  # type: MutableMapping[str, ZhaDeviceEntry]
+        self._store = hass.helpers.storage.Store(STORAGE_VERSION, STORAGE_KEY)
+
+    @callback
+    def async_create(self, device) -> ZhaDeviceEntry:
+        """Create a new ZhaDeviceEntry."""
+        device_entry = ZhaDeviceEntry(
+            name=device.name,
+            ieee=str(device.ieee),
+            power_source=device.power_source,
+            manufacturer_code=device.manufacturer_code
+
+        )
+        self.devices[device_entry.ieee] = device_entry
+
+        return self.async_update(device)
+
+    @callback
+    def async_get_or_create(self, device) -> ZhaDeviceEntry:
+        """Create a new ZhaDeviceEntry."""
+        ieee_str = str(device.ieee)
```

Comment:
Why does this need to be cast to str? That seems like something the caller of this function should take care of

Reply:
I think it’s cleaner to have the consistency of the device being the arg to all of the functions. Is there any reason I can’t leave it like this?

Reply:
Actually it's fine, since ieee is EIU64 type